armada/armada.py

```diff
@@ -111,12 +117,14 @@ def show_diff(self, chart, installed_chart,
         if len(chart_diff) > 0:
             LOG.info("Chart Unified Diff (%s)", chart.release_name)
             for line in chart_diff:
-                print line
+                LOG.debug(line)
         values_diff = list(difflib.unified_diff(installed_values.split('\n'),
                                                 yaml
                                                 .safe_dump(target_values)
                                                 .split('\n')))
         if len(values_diff) > 0:
             LOG.info("Values Unified Diff (%s)", chart.release_name)
             for line in values_diff:
-                print line
+                LOG.debug(line)
+
+        return (len(chart_diff) > 0) and (len(chart_diff) > 0)
```

Comment:
Really an upgrade would be triggered by either chart differences or value differences.

This line:

```
return (len(chart_diff) > 0) and (len(chart_diff) > 0) 
```

Looks at chart_diff twice.  Likely a typo, and meant chart_diff, values_diff but this should be an `or` rather then an `and`